ledger-tool/src/main.rs

```diff
@@ -920,6 +920,40 @@ fn open_genesis_config_by(ledger_path: &Path, matches: &ArgMatches<'_>) -> Genes
     open_genesis_config(ledger_path, max_genesis_archive_unpacked_size)
 }
 
+/// Finds the accounts needed to replay slots `snapshot_slot` to `ending_slot`.
+/// Removes all other accounts from accounts_db, and updates the accounts hash
+/// and capitalization. This is used by the --minimize option in create-snapshot
+fn minimize_bank_for_snapshot(
+    blockstore: &Blockstore,
+    bank: Arc<Bank>,
+    snapshot_slot: Slot,
+    ending_slot: Slot,
+) {
+    let (mut minimized_account_set, transaction_accounts_measure) = measure!(
+        blockstore
+            .get_accounts_used_in_range(snapshot_slot, ending_slot)
+            .unwrap(),
+        "get transaction accounts"
+    );
+    let total_accounts_len = minimized_account_set.len();
+    info!("Added {total_accounts_len} accounts from transactions. {transaction_accounts_measure}");
+
+    minimized_account_set.extend(bank.feature_set.active.iter().map(|(pubkey, _)| *pubkey));
+    minimized_account_set.extend(bank.feature_set.inactive.iter().cloned());
+    minimized_account_set.extend(solana_sdk::sysvar::ALL_IDS.iter().cloned());
+    minimized_account_set.extend(solana_runtime::builtins::get_pubkeys());
+    minimized_account_set.insert(solana_sdk::bpf_loader::id());
+    minimized_account_set.insert(solana_sdk::bpf_loader_deprecated::id());
+    minimized_account_set.insert(solana_sdk::bpf_loader_upgradeable::id());
+    minimized_account_set.insert(solana_sdk::ed25519_program::id());
+    minimized_account_set.insert(solana_sdk::secp256k1_program::id());
+    minimized_account_set.insert(solana_runtime::inline_spl_token::native_mint::id());
```

Comment:
I worry about this list potentially getting out of date. Obviously stuff like `bank.feature_set.active.iter()` should pull in new additions automatically, but additions of a new program to `solana_sdk` would seemingly not.

I'm wondering if there is a way that we could guarantee this list is up to date with a unit test

Reply:
I haven't been able to think of a good way to do this. If we created some list of these sdk programs we need in a module somewhere, it seems that could just get out of date as well.

Unless we had a way to get all the sdk program ids when registering them w/ the macro?, but I'm not sure that's possible.